### DISCUSSION TITLE
YARP as Forward Proxy with HTTPS

### DISCUSSION THREAD

**Discussion by lisandrogreco** [dotnet/yarp#1829] [Q&A] [answered]
Hi,
Not sure if this is redundant with #1637 but we try to use YARP as a forwarding Proxy. We configured an HTTP Forwarder. It works well when we call an URL with HTTP but not with HTTPS calls. 

We set our YARP endpoint as Proxy at the HTTPClient and try to call another Service that listens on HTTPS. 

When testing with cURL we got this message:
```
curl -x localhost:7285 'https://localhost:7002/api/status'
curl: (56) Received HTTP code 400 from proxy after CONNECT

curl -x localhost:7289 'https://localhost:7002/api/status'
curl: (56) Proxy CONNECT aborted
```

Port: 7285 is HTTP and 7289 is HTTPS of our YARP Proxy.
Is this even Possible to use YARP as Forward Proxy with HTTPS?

BR

**Comment by MihaZupan** [comment, marked as answer]
No, YARP is a reverse proxy and it does not implement CONNECTs.
You lose all benefits of a reverse proxy when tunneling https traffic.

See https://github.com/microsoft/reverse-proxy/discussions/298#discussioncomment-33796.

You can either
- use a forward proxy or
- point the client at YARP instead of the server and use YARP as a reverse proxy
